lib/rouge/lexers/hql.rb

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    load_lexer 'sql.rb'
+
+    class HQL < SQL
+      title "HQL"
+      desc "Hive Query Language SQL dialect"
+      tag 'hql'
+      filenames '*.hql'
+
+      def self.keywords
+        # source: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL
+        @keywords ||= Set.new(%w(
```

Comment:
In Hive, are standard SQL keywords no longer valid? This list has some overlap from the `sql.rb` keywords. For example, things like `ABORT`, `ABS`, `ABSOLUTE` are not present in the HQL keywords here. I'm only asking to make sure we're not missing some keywords and if we should do a merge of SQL keywords plus HQL keywords.

Reply:
`sql.rb` also doesn't specify the standard SQL keyword 'types', like `CHAR`, `VARCHAR`, etc, which overlap with HQL keyword types. I am kind of leaning toward specifying most of these in `sql.rb` and adding keyword types to the standard SQL lexer and then adding to it in this lexer, if necessary. It would make the rules in this lexer a lot simpler and wouldn't be as repetitive. Happy to help iron this out if you need help.

Reply:
Hey @dblessing , thanks for your comments!

I like the suggestion of moving
```ruby
rule /\w[\w\d]*/ do |m|
         if self.class.keywords_type.include? m[0].upcase
            token Keyword::Type
          elsif self.class.keywords.include? m[0].upcase
            token Keyword
          else
            token Name
          end
       end
end
```
to `ruby.rb`. Then I will also add a default list of (type) keywords there. How does that sound?

As for subclassing for hql, I'd argue that it's more maintainable to just override the full return value of `keywords` and `keywords_type`. The way I see it, the supposed advantage of only adding the delta is that any updates to sql would automatically be inherited by hql; but that's actually a _disadvantage_ as the hql spec is just the authoritative source of truth on what the keywords are. What do you think?

(Commenting from my personal account @tkluck , not by work account @tkluck-booking, don't let that confuse you.)

Reply:
As for the specific examples, `abort` is indeed not a keyword. `abs` and `absolute` are indeed an omission: I didn't copy the list of functions [as listed here][functions]. Good catch!

As per my previous comment, I have a preference for adding the full list to `hql.rb` instead of trying to maintain a delta between the two. I'm thinking of adding a method `keywords_function` to the sql class, and overriding it for the hql class. How's that?

[functions]: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+UDF